utils/s2n_result.h

```diff
@@ -41,6 +41,9 @@ S2N_RESULT_MUST_USE bool s2n_result_is_ok(s2n_result result);
 /* returns true when the result is S2N_RESULT_ERROR */
 S2N_RESULT_MUST_USE bool s2n_result_is_error(s2n_result result);
 
+/* ignores the returned result of a function */
+void s2n_result_ignore(s2n_result result);
```

Comment:
If the caller wants to signal they can do so in the callback. I don't think it's worth having our own signal for this one-off debugging feature

Reply:
In that case since we're introducing this method, can we put a scary warning about using it in the comment? Like "Generally results should not be discarded. Using this method could cause you to miss failures that should be handled". Like an "ARE YOU SURE?!" for the next developer.

Reply:
Yeah that's probably a good idea